makeflow/src/makeflow.c

```diff
@@ -2065,6 +2096,21 @@ if (enforcer && wrapper_umbrella) {
 
 	if(local_queue)
 		batch_queue_delete(local_queue);
+
+	/*
+	 * Set the abort and failed flag for batch_job_mesos mode.
+	 * Since batch_queue_delete(struct batch_queue *q) will call
+	 * batch_queue_mesos_free(struct batch_queue *q), which is defined 
+	 * in batch_job/src/batch_job_mesos.c. Then this function will check 
+	 * the abort and failed status of the batch_queue and inform 
+	 * the makeflow mesos scheduler. 
+	 */
+
+	if (batch_queue_type == BATCH_QUEUE_TYPE_MESOS) {
```

Comment:
Since Mesos run each task with an independent executor, if user abort a workflow that contains many long running tasks, it'd be better to notice Mesos to stop executors work for this workflow.  I am not sure how do other batch job systems abort running tasks, but maybe we should notice other batch job systems about failure as well.